pywbem/mof_compiler.py

```diff
@@ -2178,6 +2191,41 @@ def CreateClass(self, *args, **kwargs):
             self.classes[self.default_namespace] = \
                 NocaseDict({cc.classname: cc})
 
+        objects = list(cc.properties.values())
+        for meth in cc.methods.values():
+            objects += list(meth.parameters.values())
+
+        for obj in objects:
+            # Validate that reference_class exists in repo
+            if obj.type == 'reference':
+                try:
+                    self.GetClass(obj.reference_class, LocalOnly=True,
+                                  IncludeQualifiers=True)
+                except CIMError as ce:
+                    if ce.args[0] == CIM_ERR_NOT_FOUND:
+                        ce.args = (CIM_ERR_INVALID_PARAMETER,
+                                   'Reference object "%s" of class "%s" '
+                                   'reference_class "%s" does not exist' %
```

Comment:
Please improve the exception message to be more readable. How about: "The class referenced by property %r of class %r does not exist: %r".

Reply:
Why use %r. that may spray out extra data when all we want is the name. Otherwise no issue with message.

Reply:
The %r puts quotes around the message and the string form does not do that.

Reply:
We need to create a new issue to assure that the above is correct and that we fix it.

Reply:
DONE